lib/parser.mly

```diff
@@ -110,10 +110,66 @@ stmt_list:
 
 stmt:
   | expr SEMI                               { Expr $1      }
-  | LBRACE stmt_list RBRACE                 { Block $2 }
-  | IF LPAREN expr RPAREN stmt ELSE stmt    { If($3, $5, $7) }
-  | WHILE LPAREN expr RPAREN stmt           { While ($3, $5)  }
+  | logic_expr                              { Expr $1 }
+  | if_stmt                                 { $1 }
+  | block                                   { $1 }
+  | loop                                    { $1 }
   | RETURN expr SEMI                        { Return $2      }
+/* TODO 
+  | BREAK
+  | CONTINUE
+*/
+
+logic_expr: /* within control flow, we dont want SEMI involved in our exprs */
+  | expr { $1 }
+
+loop:
+  | WHILE expr stmt           { While ($2, $3)  }
+  | INFINITE_LOOP stmt        { While (BoolLit(true), $2)  }
+  /* for counter in starting_num ... <ending condition>ending_num step step_number */
+  | FOR ID_VAR IN LIT_INT DOT DOT DOT end_condition LIT_INT increment stmt
```

Comment:
`DOT DOT DOT` should be parsed as `SPREAD` token. otherwise like `. .\n.` will be accepted.

Reply:
oh thats a good call, what is spread? is this a new token to be added?

Reply:
yes, I believe so, `SPREAD` or `RANGE` sounds like a good token name for this.

Reply:
I will change it